### PR TITLE
simple-mpc: fix circular import

### PR DESCRIPTION
**Pull request by seanfarley**
It turned out to be a simple move of one function so I went ahead and
did that.

### CODE REVIEW

**Comment by seanfarley** [comment]
Turns out a few beers helped free up some time this weekend ^_^

File: simple-mpc.el [hunk 1]
```diff
@@ -123,36 +123,5 @@ command."
...
-              "      * [C]lear current playlist\n"
-              "      * [S]huffle playlist\n"
-              "      * [l]oad playlist\n"
-              "      * [s]earch database\n"
-              (propertize "\n   * misc\n" 'face 'simple-mpc-main-headers)
-              "      * [q]uit")
-      (simple-mpc-mode) ; start major mode
-      (switch-to-buffer buf))))
-
 (provide 'simple-mpc)
```
Comment (jorenvo) [inline simple-mpc.el:126]: We should move this `provide` as well. According to [the documentation](https://www.gnu.org/software/emacs/manual/html_node/elisp/Named-Features.html) the file that defines the symbol should provide it.

**Review by jorenvo — CHANGES_REQUESTED** [review]
Indeed, I think this should work. Just one small change is necessary.

File: simple-mpc.el (hunk 1 above)
Comment (seanfarley) [reply simple-mpc.el:126]: Ah, good catch; thanks! I'll re-read the code changes and also the docs :-)

Comment (seanfarley) [reply simple-mpc.el:126]: I might rethink my solution a bit to see if there's a better route; I'll try to do some tonight.

**Comment by seanfarley** [comment]
After looking at a few other emacs' packages, my question is why does `simple-mpc` have `simple-mpc-mode.el`? Why not just put that into `simple-mpc.el`? It seems lots of projects do it that way (including this projects inspiration, `mu`).

**Comment by jorenvo** [comment]
`simple-mpc.el` contains the main entrypoint for most users: the `simple-mpc` function. But it's not the only way to use this package, you can also use `simple-mpc-query.el` or `simple-mpc-current-playlist.el` directly by using their respective minor modes. Both of these rely on `simple-mpc-mode`, but not on the `simple-mpc` function. I think that's why I chose this file layout, but it's been a while since I wrote this :D.

I'm not completely against changing the file layout, but I think a lower-complexity fix may be better here. Regardless of what we do, we must make sure not to break existing installations (users will have `(require 'simple-mpc)` in their init).

**Comment by seanfarley** [comment]
What I was thinking was to merge `simple-mpc-mode.el` into `simple-mpc.el` because there is an inherit dependency there (`simple-mpc-mode` calls `simple-mpc`) and I'm not sure keeping them apart buys much?

Since `simple-mpc-{query,current-playlist}.el` already depend on `simple-mpc` this just adds two more definitions from their point of view. Also, I think that means this current approach in this PR won't work since I tried to move that function.

**Comment by seanfarley** [comment]
Not that this is definitive, but it doesn't seem anyone directly calls `(require 'simple-mpc-mode)` except for this package itself. So, I don't _think_ we'd be breaking any existing packages:

https://github.com/search?p=3&q=%22require+%27simple-mpc-mode%22&type=Code

**Comment by seanfarley** [comment]
Oh, I see. I misunderstood your comment. I got it backwards. Sorry for the confusion; I'll think on this some more.

**Pull request closed without merging**